packages/contracts/test/core/dao/dao.ts

```diff
@@ -1198,90 +1225,175 @@ describe('DAO', function () {
-        );
-    });
+        .withArgs();
 
-    it('should set validator and emits event', async () => {
-      const validatorAddress = ethers.Wallet.createRandom().address;
-      const tx = await dao.setSignatureValidator(validatorAddress);
+      // Grant the permission to validate signatures to the caller unconditionally
+      await dao.grant(
+        dao.address,
+        caller.address,
+        PERMISSION_IDS.VALIDATE_SIGNATURE_PERMISSION_ID
+      );
 
-      expect(await dao.signatureValidator()).to.be.eq(validatorAddress);
+      // The caller can validate signatures now.
+      expect(await dao.connect(caller).isValidSignature(hash, goodSignature)).to
+        .not.be.reverted;
 
-      await expect(tx)
-        .to.emit(dao, EVENTS.SignatureValidatorSet)
-        .withArgs(validatorAddress);
+      // Because the caller is allowed unconditionally, the validation is always true.
+      expect(
+        await dao.connect(caller).isValidSignature(hash, goodSignature)
+      ).to.equal(VALID_ERC1271_SIGNATURE);
+      expect(
+        await dao.connect(caller).isValidSignature(hash, badSignature)
+      ).to.equal(VALID_ERC1271_SIGNATURE);
     });
 
-    it('should call the signature validator', async () => {
-      const ERC1271MockFactory = await smock.mock('ERC1271Mock');
-      const erc1271Mock = await ERC1271MockFactory.deploy();
+    it('allows caller specific signature validation', async () => {
+      const caller = signers[0];
+      const signer = signers[1];
 
-      await dao.setSignatureValidator(erc1271Mock.address);
-      await dao.isValidSignature(ethers.utils.keccak256('0x00'), '0x00');
-      expect(erc1271Mock.isValidSignature).has.been.callCount(1);
+      const message = 'Hi!';
+      const hash = ethers.utils.hashMessage(message);
+      const goodSignature = await signer.signMessage(message);
+      const badSignature = await signer.signMessage('Bye!');
+      expect(badSignature).to.not.equal(goodSignature);
+
+      // Try to call with caller but caller has no permission
+      expect(await dao.connect(caller).isValidSignature(hash, goodSignature))
+        .to.be.revertedWithCustomError(dao, 'Unauthorized')
+        .withArgs();
+
+      // Grant the permission to validate signatures to the caller unconditionally
+      await dao.grant(
+        dao.address,
+        caller.address,
+        PERMISSION_IDS.VALIDATE_SIGNATURE_PERMISSION_ID
+      );
+
+      // The caller can validate signatures now.
+      expect(await dao.connect(caller).isValidSignature(hash, goodSignature)).to
+        .not.be.reverted;
+
+      // Because the caller is allowed unconditionally, the signature is always valid.
+      expect(
+        await dao.connect(caller).isValidSignature(hash, goodSignature)
+      ).to.equal(VALID_ERC1271_SIGNATURE);
+      expect(
+        await dao.connect(caller).isValidSignature(hash, badSignature)
+      ).to.equal(VALID_ERC1271_SIGNATURE);
     });
```

Comment:
This test is a duplicate of the previous one

Reply:
Fixed in 25b7704e70b50f8b8a546560083e551d0aeab194